pruner/checkpoint.go

```diff
@@ -91,5 +107,23 @@ func (s *Service) updateCheckpoint(
 }
 
 func (s *Service) lastPruned(ctx context.Context) (*header.ExtendedHeader, error) {
-	return s.getter.GetByHeight(ctx, s.checkpoint.LastPrunedHeight)
+	tail, err := s.hstore.Tail(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	lastPruned := s.checkpoint.LastPrunedHeight
+	if tail.Height() < lastPruned {
+		return s.hstore.GetByHeight(ctx, lastPruned)
+	}
+
+	s.checkpoint.LastPrunedHeight = tail.Height()
+	for height := range s.checkpoint.FailedHeaders {
+		if height < tail.Height() {
+			delete(s.checkpoint.FailedHeaders, height)
+			log.Warnw("unfailing unavailable header; some block data may not be pruned", "height", height)
```

Comment:
this is confusing message to user, maybe we can elaborate like 
```suggestion
			log.Warnw("attempt to prune block data from unavailable header; some block data may not be pruned", "height", height)
```